paddle/optimizer/parameter_optimizer.cc

```diff
@@ -0,0 +1,67 @@
+#include "parameter_optimizer.h"
+#include "optimizer_factory.h"
+#include <glog/logging.h>
+
+namespace paddle {
+namespace optimizer {
+
+template <class T>
+ParameterOptimizer<T>* ParameterOptimizer<T>::create(
+    const ::std::string &config_proto) {
+  paddle::OptimizerConfig config;
+  CHECK(config.ParseFromString(config_proto) == 0) << "error : optimizer config";
+  CHECK(config_valid(config) == 0) << "error : invalid optimizer config ";
+  ParameterOptimizer<T> *opt = nullptr;
+  switch (config.optimizer_name()) {
+  case "SGD" : opt = new SGDOptimizer<T>(config); break;
+  case "Adagrad" : opt = new AdagradOptimizer<T>(config); break;
+  case "Adadelta" : opt = new AdadeltaOptimizer<T>(config); break;
+  case "Adam" : opt = new AdamOptimizer<T>(config); break;
+  default:
+    opt = new SGDOptimizer<T>(config);
+  }
+  return opt;
+}
+
+template<class T>
+double ParameterOptimzier<T>::get_learning_rate() {
```

Comment:
As we discussed during yesterday's meeting, we plan to use constant learning rate for now. And in future we may use composition rather than inheritance. So we probably don't need to have this function `get_learning_rate` in the base class.